### ISSUE TITLE
FEAT: Improve malicious package alerts and triage 

### ISSUE THREAD

**Issue by Jayy001** [vipyrsec/bot#242]
This is the new embed I'd like to propose:
![image](https://github.com/vipyrsec/bot/assets/56607897/0d0d4935-50fb-4c6f-be25-87a083173de8)

# General idea
When getting an alert, the steps would be
* Initial reporter has a look over in the inspector, if its a false positive then they can click the "Approve" button which will change the footer text to `Approved by @user <timestamp>`. This will also change the embed colour to **green.** 
* If the reporter has some doubts or its not immediately obvious, they can click `Add Notes` to record their findings, which will add it to the `Event Log` section. For example, `Unknown URL being requested ~ @user <timestamp>` - note everything after `~` is automatically added. Also, the embed colour gets changed to **orange**.
* Any other reporter can then view these logs, and once come to a conclusion. If it looks safe they can just click the approve button. If it's malicious then they click "Report", go through the reporting process and the embed colour turns red. It might also be worth recording this to another chanel?

**Comment by import-pandas-as-numpy** [comment]
Since packages are 'rolling', in that-- a package may be questionable but not overtly malicious at first glance, this will also necessitate database changes, as we'd want these comments to carry forward to subsequent versions of the package probably. 

At the moment, we do these ad-hoc, by generating a thread for a package that requires analysis, completing the analysis (And either reporting or ignoring the package) and then simply leaving the thread to die. This is actually fairly intuitive, and often anything that requires an extensive amount of discussion _does_ end up being malicious, so we haven't really had any reasons to create a thread, conclude the analysis, and then carry forward those comments because the package wasn't "malicious enough". 

Reports _are_ recorded to another channel (#reporting), by time, date, report reason, etc. It should generally mirror the type of information we expect PyPI administrators to need to action our reports .

I am generally _for_ the idea of approving a package, but this is going to be something we're going to want to expand in the future-- an approved package probably doesn't need to ping on the same heuristics again. I'm not sure what that's going to look like though. 

TL;DR: 
- I am for the idea of changing our detection to include the specific match location in the script.
- There are some data considerations for this, specifically how we want to retain and handle the data (to include analyst comments moving forward). This will be @sid-maddy and @jonathan-d-zhang running point on... what this data retention might look like. 
- I am for the idea of Approving a given package, mostly to make it easier to communicate to others that you've read a package, but also, we'll probably expand on this feature in the future, so it's likely that this presents a good place for us to start at. 
- @Robin5605 tagging because you'd probably be overseeing this.

**Comment by Robin5605** [comment]
I like these proposed changes, some thoughts, however:
- I'm also in favor of showing _where_ a certain rule matched, sadly this is currently not possible as we do not store this information (however, it is on the roadmap: https://github.com/vipyrsec/dragonfly-mainframe/issues/227). This can be omitted for now, and added in later.
- Adding notes and marking something as viewed is great, however as Rem has pointed out, this won't be persisted across versions since it'd need database support. I think we can still go forward, and add this in later.

In general, I'm in favor with going forward with these features now if you'd like, but leaving out the parts that require database support and simply adding these in later. I feel like this may be a better course of action than blocking for database support and potentially turning this into a stale issue.

**Comment by Jayy001** [comment]
Perfect. I'll get working on it now.
